Fix mochad issue for RPi and CM19A. Recycle socket on new command to …

aosadchyy opened the pull request:
Minor fix to re-connect the underlying socket in controller.PyMochad

There is a known issue for mochad to get locked up occasionally. e.g. as seen often on RPi.
https://sourceforge.net/p/mochad/discussion/1320003/thread/1c193f01/

When this happens, mochad can be recovered by restarting it. However, mochad.py components in home-assistant once connected on HA startup, won't ever reconnect.

This improvement allows to reconnect underlying socket in controller. PyMochad when X10 commands are sent to mochad. It's tested to work fine on my HA.

balloob committed:
Version bump to 0.58.0

andrey-git committed:
Print entity type in "too slow" warnings (#10641)

* Update entity.py

* Update entity.py

robbiet480 committed:
Alexa improvements (#10632)

* Initial scene support

* Initial fan support

* ordering

* Initial lock support

* Scenes cant be deactivated; Correct the scene display category

* Initial input_boolean support

* Support customization of Alexa discovered entities

* Initial media player support

* Add input_boolean to tests

* Add play/pause/stop/next/previous to media player

* Add missing functions and pylint

* Set manufacturerName to Home Assistant since the value is displayed in app

* Add scene test

* Add fan tests

* Add lock test

* Fix volume logic

* Add volume tests

* settup -> setup

* Remove unused variable

* Set required scene description as per docs

* Allow setting scene category (ACTIVITY_TRIGGER/SCENE_TRIGGER)

* Add alert, automation and group support/tests

* Change display categories to match docs

* simplify down the display category props into a single prop which can be used on any entity

* Fix tests to expect proper display categories

* Add cover support

* sort things

* Use generic homeassistant domain for turn on/off

cezarsa committed:
Make MQTT reconnection logic more resilient and fix race condition (#10133)

bachya committed:
Fixes AirVisual bug regarding incorrect location data (#10054)

* Fixes AirVisual bug regarding incorrect location data

* Owner-requested changes

postlund committed:
Bump pyatv to 0.3.8 (#10643)

Fixes AirPlay issues on newer versions of tvOS.

balloob committed:
Update frontend to 20171118.0

robbiet480 committed:
Implement entity and domain exclude/include for Alexa (#10647)

* Implement entity and domain exclude/include for Alexa

* Switch to using generate_filter

* Use proper domain for turn on/off calls except for groups where we must use the generic homeassistant.turn_on/off

* travis fixes

* Untangle

* Lint

balloob committed:
Merge pull request #10630 from home-assistant/release-0-58

0.58

tinloaf committed:
Fix yweather (#10661)

amelchio committed:
Properly initialize Harmony remote (#10665)

The delay_secs variable was not initialized if discovery was active and no
matching configuration block existed (i.e. override was None).

etsinko committed:
Fix for time_date sensor (#10694)

* fix to time_date sensor

* cleaned up the code and added unit tests

* fixed lint errors

aosadchyy committed:
Fix mochad issue for RPi and CM19A. Recycle socket on new command to recover mochad connection

aosadchyy committed:
Fix mochad issue for RPi and CM19A. Recycle socket on new command to recover mochad connection

homeassistant commented:
Hi @aosadchyy, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10701).

Once you do that we will be able to review and accept this pull request.

Thanks!

aosadchyy committed:
Brush up the fix based on comments

homeassistant commented:
Hi @aosadchyy, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10701).

Once you do that we will be able to review and accept this pull request.

Thanks!

balloob committed:
Convert configurator to use markdown (#10668)

balloob committed:
index.html improvements (#10696)

titilambert committed:
Handle the new version of HydroQuebec website (#10682)

* Handle the new version of HydroQuebec website

* Update requirements_all.txt

titilambert committed:
Handle the new version of HydroQuebec website (#10682)

* Handle the new version of HydroQuebec website

* Update requirements_all.txt

uchagani committed:
Add Arm Custom Bypass to alarm_control_panel (#10697)

balloob committed:
Fix conversation (#10686)

* Fix conversation

* Lint

balloob committed:
Bump frontend to 20171121.0

balloob committed:
Bump frontend to 20171121.0

balloob committed:
Version bump to 0.58.1

balloob committed:
Merge pull request #10716 from home-assistant/release-0-58-1

0.58.1

balloob committed:
Merge branch 'master' into dev

balloob committed:
Shopping list: add item HTTP API (#10674)

* Shopping list: add item HTTP API

* Fix order of decorators

arsaboo committed:
Add presence device_class (#10705)

balloob committed:
Update frontend to 20171121.1

balloob committed:
Fix tests

syssi committed:
python-miio version bumped for improved device support. (#10720)

bigwoof committed:
Adding Queue count sensor (#10723)

Adding another sensor to output the numeber of items in the SABnabd queue.  This is an alternative to displaying filesize, just a preference thing, as it is more meaningfull for me the way I use SABnzdb. 

Note this is my first time coding on github and I have no idea if I am doing things right, I assume that all I needed to do is add a couple of lines to the sensors available and also another line as to what to extract from the SABnzdb API, in this case I have called the sensor "queue_count" and it gets the value from the noofslots_total which as I understand - each slot is a separate download item. 

hope I did this correctly - also I don't have a separate instance of home assistant running for testing so I have no way to test this code (and I don't know how I would switch to the dev channel either).  As I said I am a newb!

OttoWinter committed:
Bump pychromecast to 1.0.2 (#10728)

Fixes home-assistant/home-assistant#9965

bryanyork committed:
Fix unit conversion for Sensibo A/C units (#10692)

* Fix unit conversion for Sensibo A/C units

When the Sensibo component was released, there was a provision to not convert the temperature units unless "nativeTemperatureUnit" was returned with the API. I'm not sure if the API changed on Sensibo's side, but I do not get this key passed back with API requests.

This causes my current temperature to be returned in CELSIUS instead of FAHRENHEIT.

Removing this fixes it, and I can confirm the units are shown properly now.

* Update adding comment showing temperature is always in C

grischard committed:
Optimised images. Saved 80 KB out of 656 KB. 12.3% overall (up to 32.1% per file) (#10735)

svenstaro committed:
Add transmission rate (#10740)

* Add transmission rate

* Rename transmission rate attributes to shorter names

lwis committed:
Update pytradfri to 4.1.0 (#10521)

Klikini committed:
Use new DoorBirdPy (v0.1.0) (#10734)

* Use new DoorBirdPy (v0.1.0)

* Update requirements_all for DoorBirdPy 0.1.0

TD22057 committed:
Add fan and reduce I/O calls in radiotherm (#10437)

* Added fan support.  Reduced number of calls to the thermostat to a minimum

* Move temp rounding to config schema

* Fixed pep8 errors

* Fix for review comments.

* removed unneeded if block

* Added missing precision attr back

* Fixed pylint errors

* Code review fixes.  Fan support by model number.

* Defined circulate state

Rendili committed:
New Hive Component / Platforms (#9804)

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

* Changes

* Changes

* Changes

* changes

* Updates

* Updates

* Updates

* Updates

* Updates

* Updates

* Sensor code updates

* Sensor code updates

* Move sensors to binary sensors

* Quack

* Updates - Removed climate related sensors

* sensor fix

* binary_sensor updates

* New Hive Component / Platforms

* New Hive Component / Platforms

* New Hive Component / Platforms

cgtobi committed:
Temporarily fix yahoo weather API issue and add unit test. (#10737)

* Temporarily fix yahoo weather API issue and add unit test.

* Add test data.

braddparker committed:
Google assistant climate mode fix (#10726)

* Changed supported climate modes lookup to be case insensitive by forcing to lower-case

* Fixed style errors. (Blank line and line too long)

tchellomello committed:
Protect sensitive information for Amcrest cameras (#10569)

*  Creates a AmcresHub object to protect some private attributes on the logs

* Uses hass.data to pass AmcrestHub to components

* Prefer constants

* Removed serializer since it's using hass.data and simplified camera entity constructor

* small cleanup

janLo committed:
InfluxDB send retry after IOError (#10263)

* Implement data write retry for InfluxDB

This adds an optional max_retries parameter to the InfluxDB component
to specify if and how often the component should try to send the data
if the connection failed due to an IOError.

The sending will be scheduled for a retry in 20 seconds as often as the
user specified. This can be handy for flaky getwork connections between
the DB and Homeassistant or outages like daily DSL reconnects.

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Add unittest for influx write retries

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Add RetryOnError as helper decorator in util

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Add unittests for RetryOnError

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Use RetryOnError decorator in InfluxDB

This replaces the scheduling logic in the InfluxDB component with the
RetryOnError decorator from homeassistant.util

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Make the linters happy

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Implement a queue limit for the retry decorator.

This adds a queue limit to the RetryOnError handler. It limits the
number of calls waiting for be retried. If this number is exceeded,
every new call will discard the oldest one in the queue.

* influxdb: Add the retry queue limit option.

* Make the linter happy.

* Make pylint happy

* Log exception of dropped retry

* Move RetryOnError decorator to influxdb component.

* Fix bug in logging usage

* Fix imports

* Add newlines at the end of files.

* Remove blank line

* Remove blank line

craigjmidwinter committed:
Add Dominos Pizza platform (#10379)

* add dominos service

* change require

* dump to log

* component fixes

* clean-up use updated library

* remove unnecessary import

* fix hound errors

* more lint fixes

* Coverage rc

* update requirements

* cleanup as per notes

* missing message

* linting...

* schema validation and reducing requests

* fixlint

* spacing

* unused variable

* fix docstrings

* update req

* notes updates, pypi package, front-end panel

* stale import

* fix constant name

* docstrings

* fix library import

* lint fixes

* pylint bug

* remove built-in panel

* Make synchronous

* unused import and use throttle

* Handle exceptions properly and update client

* Import exceptions properly

* unused import

* remove bloat from start-up, readability fixes from notes, retrieve menu on request, not on startup

* whitespace on blank line

zBart committed:
Fix name collision when using multiple Hue bridges (#10486)

* Fix name collision when using multiple Hue bridges

See https://github.com/home-assistant/home-assistant/issues/9393

* Use new style of string formatting

* Removed creating of "All Hue Lights" group

n8henrie committed:
Make shell_command async (#10741)

* Make shell_command async

Use `asyncio.subprocess` instead of `subprocess` to make the
`shell_command` component async.

Was able to migrate over existing component and tests without too many
drastic changes.

Retrieving stdout and stderr paves the way for possibly using these in
future feature enhancements.

* Remove trailing comma

* Fix lint errors

* Try to get rid of syntaxerror

* Ignore spurious pylint error

Rendili committed:
Update CODEOWNERS with hive Component / Platforms (#10775)

uchagani committed:
Bump total-connect-client version (#10769)

balloob committed:
Fix scene description formatting. (#10785)

uchagani committed:
Fixes #10773: Demo Alarm Broken (#10777)

* Fixes #10773: Demo Alarm Broken

* Added test for platform setup

* Remove unused import

* Lint fix

* Rework assert to work with python 3.5

tchellomello committed:
Load Ring camera only with Ring Protect plan activated (#10739)

* Added ability to only load Ring camera
if the Ring Protect plan is activated.

* Fixed notification for all invalid cameras

* Fixed attribute name

* Using asyncio for persistent notifications

andrey-git committed:
system_log improvements (#10709)

* system_log improvements

* Don't use ModuleNotFoundError which is 3.6+

* Don't use FrameSummary which was added in 3.5

* Don't trace stack for exception logs

* Handle test error in Python 3.4

milanvo committed:
Fix WUnderground error handling, rework entity methods (#10295)

* WUnderground sensor error handling and sensor class rework

* WUnderground error handling, avoid long state, tests

* Wunderground - add handling ValueError exception on parsing

* Changes to address review comments - part 1

* Tests lint

* Changes to address review comments - part 2

bcl1713 committed:
Add away_mode_name to arlo alarm control panel (#10796)

* Update arlo.py

Include variables for custom away mode specification

* fixed line too long style problem

* fix trailing white space

* fix sending away mode command

balloob committed:
Bump frontend to 20171126.0

fabaff committed:
Add missing docstring (#10812)

* Add missing docstring

* Revert isort change

balloob committed:
Bump frontend to 20171127.0

balloob committed:
Cloud cognito switch (#10823)

* Allow email based cognito instance

* Fix quitting Home Assistant while reconnecting

* Lint

rasmusbe committed:
Added sensor types from telldus server src (#10787)

Added from https://github.com/telldus/tellstick-server/blob/master/telldus/src/telldus/Device.py

zhujisheng committed:
Add tts.baidu platform (#10724)

* Add tts.baidu platform

* Update baidu.py

* changed to sync

get_engine and get_tts_audio changed to sync.

perosb committed:
make RGB values consistent as int. fixes #10766 (#10782)

* make RGB consitant as int. fixes #10766

* fix rounding and only change for hex convertion

matemaciek committed:
Single LEDs in Blinkt support (#10581)

* Single LEDs in Blinkt support

* Review remarks

dgomes committed:
Add debug (#10828)

stephenyeargin committed:
Fix "recently pair device" (#10832)

Noticed a minor grammar mistake.

chocomega committed:
Fixed Yeelight's color temperature conversion to RGB (#10831)

andrey-git committed:
Fix for Sensibo with missing temperature (#10801)

* Fix for sensibo woth missing temperature

* Use new temperatureUnit API field

dferrante committed:
upgrade somecomfort to 0.5.0 (#10834)

* upgrading somecomfort to 0.5.0

* upgrade somecomfort to 0.5.0 in requirements files

odinuge committed:
Add support for logarithm in templates (#10824)

* Add support for logarithm in templates

This adds a 'log' filter that takes the logarithm of the given value,
with an optional base number. The base defaults to 'e' - the natural
logarithm

* Remove usage of log10 in template filter 'log'

* Add logarithm as a global

This makes it possible to write:
'{{ log(4, 2) }}'

snowzach committed:
Changing handling for google_assistant groups to treat them as lights. (#10111)

* Fixed aliases warning message

* Fixed test cases

* Changing handling for google_assistant groups to treat them as lights - amending to include user info.

* Enable brightness, RGB, etc for groups in Google Assistant

* Revert color/hue/temp settings

* Change servce from light to homeassistant

* Fixed config_units

* Convert from light to switch

* Change group to switch

* Update tests to switch instead of light for group

fredrike committed:
Tellduslive update with support for auto config and Local api (#10435)

* Add support for local api connection found in TellStick Znet Lite/Pro.
Added auto discovery support for all TelldusLive devices,
changed authentication method. Breaking change!
Upgraded tellduslive dependency
Update CODEOWNERS.

* Close any open configurator when configuration done

* Add support for Telldus Local API via config (#2)

* Updated dependency (addresses issue raised by @rasmusbe in https://github.com/home-assistant/home-assistant/pull/10435#issuecomment-344719714)

* Fix requested changes

Julius2342 committed:
KNX: Added config option for broadcasting current time to KNX bus. (#10654)

balloob commented on homeassistant/components/switch/mochad.py:
```diff
@@ -50,24 +50,46 @@ def __init__(self, hass, ctrl, dev):
         self._comm_type = dev.get(mochad.CONF_COMM_TYPE, 'pl')
         self.device = device.Device(ctrl, self._address,
                                     comm_type=self._comm_type)
-        self._state = self._get_device_status()
```
Why no longer fetch the state on startup?

aosadchyy replied:
The original development must have been done with cm15a (has both pl and rf). I'm using cm19a (has only rf), which cannot get the status for power line switches. It delays the startup for ca 50 seconds on 10 devices.

aosadchyy replied:
Ok. fixed in commit 3. e88301b

aosadchyy replied:
I've made the old and new line conditional. See commit 5. In case of comm_type == 'pl' the status will be retrieved as previously. Both cm15a and cm19a should be fine now.

balloob commented on homeassistant/components/switch/mochad.py:
```diff
@@ -50,24 +50,46 @@ def __init__(self, hass, ctrl, dev):
-        self._state = self._get_device_status()
+        """Init with false to avoid locking HA for long on CM19A"""
+        self._state = False
+        self.lock = threading.Lock()
```
Should this be a lock shared among all devices?

aosadchyy replied:
good point. I will need to move the lock creation up to the setup_platform(), then pass by reference to all devices.

aosadchyy replied:
Ok. fixed in commit 3. e88301b

balloob commented on homeassistant/components/switch/mochad.py:
```diff
@@ -50,24 +50,46 @@ def __init__(self, hass, ctrl, dev):
+            self.lock.acquire()
+            """Turn the switch on."""
+            self._state = True
+            """Recycle socket on new command to recover mochad connection"""
```
Use `#` for comments except for the python docs as first entry in a method.

aosadchyy replied:
Ok. fixed in commit 3. e88301b

balloob commented on homeassistant/components/switch/mochad.py:
```diff
@@ -50,24 +50,46 @@ def __init__(self, hass, ctrl, dev):
-        self.device.send_cmd('off')
-        self._controller.read_data()
+        try:
+            self.lock.acquire()
```
Use `with` blocks to acquire/release locks.

```python
with self.lock:
    try:
        …
```

aosadchyy replied:
Ok. fixed in commit 3. e88301b

cbulock committed:
Ecobee set humidity level (#10780)

* Add the ability to set humidity levels on ecobee thermostats

* use the latest version of python-ecobee-api

* Lint fixes

schmittx committed:
Add device class for low battery (#10829)

molobrakos committed:
Tellduslive: Use magic constants for battery level. Also, the previous formula for battery level was wrong. (#10788)

andrey-git committed:
Add useragent-based detection of JS version (#10776)

* Add useragent-based detection of JS version

* Keep es5 as default meanwhile

* Update test

perosb committed:
add support for multiple execution per execute request (#10844)

danielperna84 committed:
Upgrade HomeMatic, add devices (#10845)

tinloaf committed:
Climate component: add supported_features (#10658)

* Implement supported_features for the climate component

* Test supported features

* Convert generic thermostat to supported features

* Max / min temperature are not features

* Fix lint

* Min / max humidity are not features

* Linting

* Remove current temperature / humidity

* Move c-hacker-style constants to boring integers. Booo!

* Refactor all the climate platforms to use the new supported_features

* Force all climate platforms to implement supported_features

* Fix mistakes

* Adapt hive platform

* Move flags into a constant

* Calm the hound

fabaff committed:
Upgrade mutagen to 1.39 (#10851)

bachya committed:
Updated codeowner for Tile device tracker (#10861)

aosadchyy committed:
Addressed comments by balloob

aosadchyy committed:
typo in with self.lock:

Julius2342 committed:
Revert "KNX: Added config option for broadcasting current time to KNX bus. (#10654)" (#10874)

This reverts commit cadd797200e339e165dbeff152169e4f4d716e2d.

As discussed within #10708 we should chose a different implementation. Therefore we should revert this change to avoid a breaking change.

fabaff committed:
Upgrade distro to 1.1.0 (#10850)

pvizeli committed:
Bugfix trigger state with multible entities (#10857)

* Bugfix trigger state with multible entities

* Fix numeric state

* fix lint

* fix dict

* fix unsub

* fix logic

* fix name

* fix new logic

* add test for state

* add numeric state test for unsub

* add test for multible entities

* Update numeric_state.py

* Update numeric_state.py

* Update state.py

* Fix logic for triple match

* Add clear to numeric state

* clear for state trigger

pvizeli committed:
tellstick fix DEPENDENCIES and update tellcore-net (#10859)

* Update requirements_all.txt

* Update tellstick.py

* Fix DEPENDENCIES

* Update requirements_all.txt

* fix format

* fix lint

* fix lint

* Update tellstick.py

* update tellcore-net

* update tellcore-net

* besser validate

balloob committed:
Update frontend to 20171130.0

fabaff committed:
Upgrade aiohttp to 2.3.5 (#10889)

fabaff committed:
Upgrade fastdotcom to 0.0.3 (#10886)

fabaff committed:
Upgrade schiene to 0.19 (#10887)

rytilahti committed:
Xiaomi Vacuum: remove deprecated calls (#10839)

* vacuum.xiaomi_miio: read dnd status properly instead of using imprecise dnd flag from vacuum_state

* vacuum.xiaomi_miio: use miio package instead of mirobo

* check only that wanted calls have taken place, ignore order of calls

* Fix linting issues

* Remove empty line after docstring

PhracturedBlue committed:
Create ecobee weather platform (#10869)

* Create ecobee weather component

* Update requirements_all for ecobee

* Fix missed lint issue

jeroenterheerdt committed:
Microsoft Text-to-speech: Fixing missing en-gb support bug (#10429)

* Fixing missing en-gb support bug

* Microsoft TTS adding support for rate, volume, pitch and contour.

* Microsoft TTS fixing support for jp-jp.

* Fixing linting error on line 67

* make impossible things possible :tada:

fabaff committed:
Upgrade youtube_dl to 2017.11.26 (#10890)

fabaff committed:
Upgrade yarl to 0.15.0 (#10888)

pvizeli committed:
Fix tests (#10891)

GenericStudent committed:
Refactored WHOIS sensor to resolve assumed key errors (#10662)

* Refactored WHOIS sensor to resolve assumed key errors

Altered it to now set an attribute key and value only if the attribute
is present in the WHOIS response. This prevents assumed keys (registrar)
from raising a KeyError on WHOIS lookups that don't contain registrar
information (onet.pl, wp.pl, for example).

* Removed non-used self._data

* WHOIS sensor now creates a new local attributes dict and overrides

* Corrected typos, refactored error cases to clear state adn attributes

* Resolved double return and refactored error logging

andrey-git committed:
Serve latest extra_html in dev mode (#10863)

andrey-git committed:
Reload groups after saving a change via config API (#10877)

fabaff committed:
Version bump to 0.59.0

PhracturedBlue committed:
Update ecobee version to fix stack-trace issue (#10894)

raymccarthy committed:
Pybotvac multi (#10843)

* Update requirements_all.txt

* Update neato.py

PhracturedBlue committed:
Fix issues from review of ecobee weather component (#10903)

* Fix issues from review

* Don't use STATE_UNKNOWN

PhracturedBlue committed:
Fix issues from review of ecobee weather component (#10903)

* Fix issues from review

* Don't use STATE_UNKNOWN

turbokongen committed:
Bugfix #10902 (#10904)

turbokongen committed:
Bugfix #10902 (#10904)

fabaff committed:
Merge pull request #10898 from home-assistant/release-0-59

0.59

fabaff committed:
Merge branch 'master' into dev

bonzini committed:
More declarative timeout syntax for manual alarm control panel. (#10738)

More declarative timeout syntax for manual alarm control panel

nickovs committed:
Unpacking RESTful sensor JSON results into attributes. (#10753)

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* sensor.envirophat: add missing requirement (#7451)

Adding requirements that is not explicitly pulled in by the library
that manages the Enviro pHAT.

* PyPI Openzwave (#7415)

* Remove default zwave config path

PYOZW now has much more comprehensive default handling for the config
path (in src-lib/libopenzwave/libopenzwave.pyx:getConfig()). It looks in
the same place we were looking, plus _many_ more. It will certainly do a
much better job of finding the config files than we will (and will be
updated as the library is changed, so we don't end up chasing it). The
getConfig() method has been there for a while, but was subsntially
improved recently.

This change simply leaves the config_path as None if it is not
specified, which will trigger the default handling in PYOZW.

* Install python-openzwave from PyPI

As of version 0.4, python-openzwave supports installation from PyPI,
which means we can use our 'normal' dependency management tooling to
install it. Yay.

This uses the default 'embed' build (which goes and downloads
statically sources to avoid having to compile anything locally). Check
out the python-openzwave readme for more details.

* Add python-openzwave deps to .travis.yml

Python OpenZwave require the libudev headers to build. This adds the
libudev-dev package to Travis runs via the 'apt' addon for Travis.

Thanks to @MartinHjelmare for this fix.

* Update docker build for PyPI openzwave

Now that PYOZW can be install from PyPI, the docker image build process
can be simplified to remove the explicit compilation of PYOZW.

* Add datadog component (#7158)

* Add datadog component

* Improve test_invalid_config datadog test

* Use assert_setup_component for test setup

* Fix object type for default KNX port

#7429 describes a TypeError that is raised if the port is omitted in the config for the KNX component (integer is required (got type str)). This commit changes the default port from a string to an integer. I expect this will resolve that issue...

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Added support for extracting JSON attributes from RESTful values

Setting the json_attributes configuration option to true on the
RESTful sensor will cause the result of the REST request to be parsed
as a JSON string and if successful the resulting dictionary will be
used for the attributes of the sensor.

* Added requirement that RESTful JSON results used as attributes must be
objects, not lists.

* Expanded test coverage to test REFTful JSON attributes with and
without a value template.

* Fixed breaks cause by manual upstream merge.

* Added one extra blank line to make PyLint happy.

* Switched json_attributes to be a list of keys rather than a boolean.

The value of json_attributes can now be either a comma sepaated list
of key names or a YAML list of key names. Only matching keys in a
retuned JSON dictionary will be mapped to sensor attributes.

Updated test cases to handle json_attributes being a list.

Also fixed two minor issues arrising from manual merge with 0.58 master.

* Added an explicit default value to the json_attributes config entry.

* Removed self.update() from __init__() body.

* Expended unit tests for error cases of json_attributes processing.

* Align quotes

fabaff committed:
Bump dev to 0.60.0.dev0 (#10912)

molobrakos committed:
Update eliqonline.py (#10914)

Channel id is now required (change in API)

RyuzakiKK committed:
Add iAlarm support (#10878)

* Add iAlarm support

* Minor fixes to iAlarm

* Rename ialarmpanel to ialarm and add a check for the host value

* corrections in the value validation of ialarm

* add a missing period on ialarm

RiRomain committed:
Correction of Samsung Power OFF behaviour (#10907)

* Correction of Samsung Power OFF behaviour

Addition of a delay after powering OFF a Samsung TV, this avoid status
update from powering the TV back ON.
Deletion of update() return statement, return value not used.

* Rename self._end_of_power_off_command into self._end_of_power_off

* Removal of unused line break in Samsung TV component

fabaff committed:
Add Alpha Vantage sensor (#10873)

* Add Alpha Vantage sensor

* Remove data object

* Remove unused vars and change return

* Fix typo

scarface-4711 committed:
Don't repeat getting receiver name on each update / pushed to denonavr 0.5.5 (#10915)

MartinHjelmare commented on homeassistant/components/switch/mochad.py:
```diff
@@ -59,15 +63,31 @@ def name(self):
+            try:
+                self._state = True
+                #Recycle socket on new command to recover mochad connection
+                _LOGGER.debug("Reconnect {}:{} ".format(self._controller.server,
```
```py
_LOGGER.debug("Reconnect %s:%s", self._controller.server, self._controller.port)
```

aosadchyy replied:
Addressed in commit 5

MartinHjelmare commented on homeassistant/components/switch/mochad.py:
```diff
@@ -59,15 +63,31 @@ def name(self):
+                self.device.send_cmd('on')
+                self._controller.read_data()
+            except Exception as e:
+                _LOGGER.error("Error with mochad communication: {}".format(e))
```
Same as above.

aosadchyy replied:
Addressed in commit 5

MartinHjelmare commented on homeassistant/components/switch/mochad.py:
```diff
@@ -59,15 +63,31 @@ def name(self):
+            try:
+                self._state = False
+                #Recycle socket on new command to recover mochad connection
+                _LOGGER.debug("Reconnect {}:{} ".format(self._controller.server,
```
Same as above.

aosadchyy replied:
Addressed in commit 5

MartinHjelmare commented on homeassistant/components/switch/mochad.py:
```diff
@@ -59,15 +63,31 @@ def name(self):
+                self.device.send_cmd('off')
+                self._controller.read_data()
+            except Exception as e:
+                _LOGGER.error("Error with mochad communication: {}".format(e))
```
Same as above.

aosadchyy replied:
Addressed in commit 5

Commit:
Add Min and Event Count Metrics To Prometheus (#10530)

* Added min and Events sensor types to prometheus

* Updated prometheus client and fixed invalid swith state

* Added metric to count number of times an automation is triggered

* Removed assumption that may not apply to everybody

* Fixed tests

* Updated requirements_test_all

* Fixed unit tests

wrboyce committed:
fix ios component config generation (#10923)

Fixes: #19020

wrboyce committed:
fix ios component config generation (#10923)

Fixes: #19020

danielperna84 committed:
Fix Notifications for Android TV (#10798)

* Fixed icon path, added dynamic icon

* Addressing requested changes

* Using DEFAULT_ICON

* Using CONF_ICON from const

* Getting hass_frontend path via import

* Lint

* Using embedded 1px transparent icon

* woof -.-

* Lint

danielperna84 committed:
Fix Notifications for Android TV (#10798)

* Fixed icon path, added dynamic icon

* Addressing requested changes

* Using DEFAULT_ICON

* Using CONF_ICON from const

* Getting hass_frontend path via import

* Lint

* Using embedded 1px transparent icon

* woof -.-

* Lint

balloob committed:
Update coveragerc (#10931)

* Sort coveragerc

* Add climate.honeywell and vacuum.xiaomi_miio to coveragerc

balloob committed:
Update frontend to 20171204.0 (#10934)

balloob committed:
Update frontend to 20171204.0 (#10934)

craigjmidwinter committed:
Dominos no order fix (#10935)

* check for none

* fix error from no store being set

* typo

* Lint

* fix default as per notes. Lint fix and make closest store None not False

* update default

craigjmidwinter committed:
Dominos no order fix (#10935)

* check for none

* fix error from no store being set

* typo

* Lint

* fix default as per notes. Lint fix and make closest store None not False

* update default

balloob committed:
Version bump to 0.59.1

balloob committed:
Merge pull request #10936 from home-assistant/release-0-59-1

0.59.1

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

aosadchyy committed:
Logger string formatting. Bug fix for CM19A.

DanNixon committed:
Report availability of TP-Link smart sockets (#10933)

* Report availability of TP-Link smart sockets

* Changes according to our style guide

nbougues committed:
Set percent unit for battery level so that history displays properly; edited variable name for consistency (#10932)

hudashot committed:
Export climate status and target temperature to Prometheus (#10919)

* Export climate metrics to Prometheus.

This adds climate_state and temperature_c metrics for each climate
device.

* Add more climate states to state_as_number

dasos committed:
Tado ignore invalid devices (#10927)

* Ignore devices without temperatures

* Typo

* Linting

* Removing return false

* Another typo. :(

* Spelling received correctly

molobrakos committed:
Upgrade tellduslive library version (closes https://github.com/home-assistant/home-assistant/issues/10922) (#10950)

molobrakos committed:
Upgrade tellduslive library version (closes https://github.com/home-assistant/home-assistant/issues/10922) (#10950)

PaulAnnekov committed:
don't ignore voltage data if sensor data changed (#10925)

mateuszdrab committed:
Fix linksys_ap.py by inheriting DeviceScanner (#10947)

As per issue #8638, the class wasn't inheriting from DeviceScanner, this commit patches it up.

mateuszdrab committed:
Fix linksys_ap.py by inheriting DeviceScanner (#10947)

As per issue #8638, the class wasn't inheriting from DeviceScanner, this commit patches it up.

Commit:
Add ADS component (#10142)

* add ads hub, light and switch

* add binary sensor prototype

* switch: use adsvar for connection

* fix some issues with binary sensor

* fix binary sensor

* fix all platforms

* use latest pyads

* fixed error with multiple binary sensors

* add sensor

* add ads sensor

* clean up after shutdown

* ads component with platforms switch, binary_sensor, light, sensor

add locking

poll sensors at startup

update state of ads switch and light

update ads requirements

remove update() from constructors on ads platforms

omit ads coverage

ads catch read error when polling

* add ads service

* add default settings for use_notify and poll_interval

* fix too long line

* Fix style issues

* no pydocstyle errors

* Send and receive native brightness data to ADS device to prevent issues with math.floor reducing brightness -1 at every switch

* Enable non dimmable lights

* remove setting of self._state in switch

* remove polling

* Revert "remove polling"

This reverts commit 7da420f82385a4a5c66a929af7025c00ed197e86.

* add service schema, add links to documentation

* fix naming, cleanup

* re-remove polling

* use async_added_to_hass for setup of callbacks

* fix comment.

* add callbacks for changed values

* use async_add_job for creating device notifications

* set should_poll to False for all platforms

* change should_poll to property

* add service description to services.yaml

* add for brigthness not being None

* put ads component in package

* Remove whitespace

* omit ads package

craigjmidwinter committed:
Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

craigjmidwinter committed:
Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

HerrHofrat committed:
Gearbest sensor (#10556)

* Added Gearbest Sensor

* Updated required files

* Fixed houndci-bout findings

* Fix tox lint errors

* Changed code according to review
Implemented library version 1.0.5

* Fixed houndci-bot findings

* Fixed tox lint issues

* Updated item schema to use has_at_least_one_key
Added conf constants

* Remove CONF_ constants and import them from homeassistant.const

* Removed CurrencyConverter from hass
Fixed couple of issues found by MartinHjelmare

b10m committed:
Add Ziggo Mediabox XL media_player (#10514)

* Add Ziggo Mediabox XL media_player

* Using pypi module ziggo-mediabox-xl now.

* Code review changes

ziotibia81 committed:
Generic thermostat initial_operation_mode (#10690)

* Generic thermostat restore operation mode

* Test restore operation mode

* Fix trailing whitespace

* Fix line too long

* Fix test duplicate entity_id

* Fix test

* async_added_to_hass modify modify internal state

* Test inital_operation_mode

* More restore state tests

* Fix whitespace

* fix test_custom_setup_param

* Test "None" target temp

emlove committed:
Use new build path for dev translations (#10937)

masarliev committed:
Add option to set default hide if away for new devices (#10762)

* Option to change hide_if_away

* tests fix

* change new device defaults

* tests and requested changes

* fix assert

Commit:
Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

Commit:
Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

balloob committed:
Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

balloob committed:
Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

andrey-git committed:
Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

andrey-git committed:
Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

balloob committed:
Version bump to 0.59.2

balloob committed:
Merge pull request #10990 from home-assistant/release-0-59-2

0.59.2

balloob committed:
Merge branch 'master' into dev

DanNixon committed:
Allow disabling the LEDs on TP-Link smart plugs (#10980)

balloob committed:
Update frontend to 20171206.0

masarliev committed:
Meraki AP Device tracker (#10971)

* Device tracker for meraki AP

* styles fix

* fix again

* again

* and again :)

* fix hide if away

* docs and optimization

* tests and fixes

* styles

* styles

* styles

* styles

* styles fix. Hope last

* clear track new

* changes

* fix accuracy error and requested changes

* remove meraki from .coveragerc

* tests and minor changes

* remove location

pvizeli committed:
Update tradfri.py (#10991)

OddBloke committed:
webostv: Ensure source exists before use (#10959)

In a case where either (a) an incorrect source name is used, or (b) the
TV isn't currently queryable (e.g. it's off), we get tracebacks because
we assume the source that we are being asked to select exists in
self._source_list.

This makes the lookup code a little more rugged, and adds in a warning
message (in place of the current exception).

Commit:
Ensure Docker script files uses LF line endings to support Docker for Windows. (#10067)

alanfischer committed:
Added Vera scenes (#10424)

* Added Vera scenes

* Fixed flake8 issues

* Fixed comments

* Moved vera to use hass.data

* Made requested changes

jeroenterheerdt committed:
Fix Egardia alarm status shown as unknown after restart (#11010)

lwis committed:
Handle OSError when forcibly turning off media_player.samsungtv (#10997)

mar-schmidt committed:
Shuffle support in Sonos (#10875)

* initial commit of shuffle option for sonos

* added test

* Small adjustments to adhere to review requests

* Removed unnessesary setting of variable. Use shuffle state from soco instead

snjoetw committed:
Add support for Canary component and platforms (#10306)

* Add Canary component

* Made some change to how canary data is updated and stored

* Updated to use py-canary:0.1.2

* Addressed flake8 warnings

* Import canary API locally

* Import canary API locally again

* Addressed pylint errors

* Updated requirements_all.txt

* Fixed incorrect unit of measurement for air quality sensor

* Added tests for Canary component and sensors

* Updated canary component to handle exception better when initializing

* Fixed tests

* Fixed tests again

* Addressed review comments

* Fixed houndci error

* Addressed comment about camera force update

* Addressed comment regarding timeout when fetching camera image

* Updated to use py-canary==0.2.2

* Increased update frequency to 30 seconds

* Added support for Canary alarm control panel

* Address review comments

* Fixed houndci error

* Fixed lint errors

* Updated test to only test setup component / platform

* Fixed flake error

* Fixed failing test

* Uptake py-canary:0.2.3

* canary.alarm_control_panel DISARM is now mapped to canary PRIVACY mode

* Fixed failing tests

* Removed unnecessary methods

* Removed polling in canary camera component and update camera info when getting camera image

* Added more tests to cover Canary sensors

* Address review comments

* Addressed review comment in tests

* Fixed pylint errors

* Excluded canary alarm_control_panel and camera from coverage calculation

amelchio committed:
Ignore Sonos players with unknown hostnames (#11013)

amelchio committed:
Support LIFX Mini products (#10996)

* Support new LIFX products

* Remove lint

todschmidt committed:
Update snips to listen on new mqtt topic and utilize rawValue (#11020)

* Updated snips to listen on new mqtt topic and use rawValue if value not present in slot

* Too late at night

* Trying to make minor changes via web

* Update test_snips.py

* Update __init__.py

* Updated wrong branch cause I'm a monkey

andrey-git committed:
Change default js version to auto (#10999)

mtreinish committed:
Serialize mochad requests (#11029)

All mochad devices are sharing a single socket interface. When multiple
threads are issuing requests to the mochad daemon at the same time the
write read cycle might get crossed between the threads. This is normally
not an issue for 1-way X10 devices because as long as the request issued
successfully and data is read over the socket then we know as much as
mochad will tell us (since there is no ACK from the request for most
X10 devices). However, where it does matter is on the device __init__()
because we're relying on the mochad daemon's internal state to take an
educated guess at the device's state to intialize things with. When
there are multiple devices being initialized at the same time the wires
can get crossed between and the wrong device state may be read.

To address this potential issue this commit adds locking using a
semaphore around all pairs of send_cmd() and read_data() (which is what
pymochad.device.Device.get_status() does internally) calls to the mochad
controller to ensure we're only ever dealing with a single request at a
time.

Fixes mtreinish/pymochad#4

andrey-git committed:
In dev mode expose only relevant sources (#11026)

foxel committed:
Added force_update for REST sensor (#11016)

* Added force_update for REST sensor

* Linting error

GreenTurtwig committed:
Add support for Logitech UE Smart Radios. (#10077)

* Add support for Logitech UE Smart Radios.

* Removed full stops to please Hound's line limit.

* Updated with requested changes.

* Fix Pylint Flake8 problem.

* Updated with requested changes.

balloob committed:
Make notify.html5 depend on config (#11052)

tringler committed:
This change fixes the error `OSError: [WinError 193]` on Windows debuggers (i.e. PyCharm) (#11034)

MartinHjelmare commented:
https://github.com/home-assistant/home-assistant/pull/11029 has now already introduced a lock for mochad. You need to rebase the branch here on upstream dev and solve the conflicts.

perfalk committed:
Added support for cover in tellstick (#10858)

* Added support for cover in tellstick

* Fixed comments from PR

* Fixed comments from PR

* Address comments

maxlaverse committed:
Add a caldav calendar component (#10842)

* Add caldav component

* Code review - 1

* Code review - 2

* Sort imports

addelovein committed:
Add GPS coords to meraki (#10998)

andreacampi committed:
Refactor hue to split bridge support from light platform (#10691)

* Introduce a new Hue component that knows how to talk to a Hue bridge, but doesn't actually set up lights.

* Refactor the hue lights platform to use the HueBridge class from the hue component.

* Reimplement support for multiple bridges

* Auto discover bridges.

* Provide some migration support by showing a persistent notification.

* Address most feedback from code review.

* Call load_platform from inside HueBridge.setup passing the bridge id.

Not only this looks nicer, but it also nicely solves additional bridges being added after initial setup (e.g. pairing a second bridge should work now, I believe it required a restart before).

* Add a unit test for hue_activate_scene

* Address feedback from code review.

* After feedback from @andrey-git I was able to find a way to not import phue in tests, yay!

* Inject a mock phue in a couple of places

uchagani committed:
add custom bypass status to total connect (#11042)

* add custom bypass status to total connect

* remove logger line

molobrakos committed:
Volvo on call: Optional use of Scandinavian miles. Also add average fuel consumption property (#11051)

pvizeli committed:
Allow tradfri to read the available state of the device (#11056)

* Allow tradfri to read the available state of the device

* Update tradfri.py

fabaff committed:
Upgrade aiohttp to 2.3.6 (#11079)

fabaff committed:
Upgrade yarl to 0.16.0 (#11078)

fabaff committed:
Upgrade psutil to 5.4.2 (#11083)

fabaff committed:
Upgrade youtube_dl to 2017.12.10 (#11080)

fabaff committed:
Upgrade shodan to 1.7.7 (#11084)

pvizeli committed:
Update tellcore-net to 0.4 (#11087)

* Update tellcore-net to 0.4

* Update requirements_all.txt

jalmeroth committed:
Bump pymusiccast to version 0.1.6 (#11091)

fabaff committed:
Use luftdaten module (#10970)

* Use luftdaten module

* Refactoring

* Check meta data

* Make name consistent

* Remove try block

postlund committed:
Bump pyatv to 0.3.9 (#11104)

DanNixon committed:
Report availability for TP-Link smart bulbs (#10976)

emosenkis committed:
Fix incorrect comment. (#11111)

balloob committed:
Update Warrant (#11101)

* Update Warrant

* Lint

ryanm101 committed:
Fixed typo in automation.py (#11116)

pschmitt committed:
Add media position properties (#10076)

* Add media progress information

* Remove unnecessary comments

* Remove datetime import

* Remove pysonyavr dependency

* Fix doc syntax (D205)

* Lint fix: no-else-return

* Don't attempt to set media progress info if program is None

* Fix Python 3.4 compatibility

* Explicitely depend on pyteleloisirs

* Only update remaining play time when it changed

* Fix floot state table

nkgilley committed:
update pyripple (#11122)

andreacampi committed:
Skip HASS emulated Hue bridges from detection. (#11128)

When refactoring the Hue support we lost a check for HASS bridges; restore that.

andreacampi committed:
Always consume the no_throttle keyword argument. (#11126)

The current code relies on the assumption that the first invocation will never specify no_throttle=True.
However that puts us in a pickle when writing unit tests: if we had a fictitious:

  def setup_platform():
    update()

  @Throttle(MIN_TIME_BETWEEN_SCANS)
  def update():
    pass

Then given multiple tests, the second and some of subsequent tests would be throttled (depending on timing).
But we also can't change that code to call `update(no_throttle=True)' because that's not currently accepted.

This diff shouldn't change the visibile behavior of any component, but allows this extra flexibility.

emlove committed:
Add problem device class (#11130)

pollett committed:
set default utc offset to 0 (#11114)

BryanJacobs committed:
Allow using more than one keyboard remote (#11061)

* Allow using more than one keyboard remote

This sets up one thread per keyboard remote, listening for events.

* Remove enclosing block in keyboard_remote

* Remove unnecessary semantic check for keyboard_remote

OverloadUT committed:
ISY994 sensor improvements (#10805)

* Fire events for ISY994 control events

This allows hass to react directly to Insteon button presses (on switches and remotes), including presses, double-presses, and long holds

* Move change event subscription to after entity is added to hass

The event handler method requires `self.hass` to exist, which doesn't have a value until the async_added_to_hass method is called. Should eliminate a race condition.

* Overhaul binary sensors in ISY994 to be functional "out of the box"

We now smash all of the subnodes from the ISY994 in to one Hass binary_sensor, and automatically support both paradigms of state reporting that Insteon sensors can do. Sometimes a single node's state represents the sensor's state, other times two nodes are used and only "ON" events are sent from each. The logic between the two forunately do not conflict so we can support both without knowing which mode the device is in.

This also allows us to handle the heartbeat functionality that certain sensors have - we simply store the timestamp of the heartbeat as an attribute on the sensor device. It defaults to Unknown on bootup if and only if the device supports heartbeats, due to the presence of subnode 4.

* Parse the binary sensor device class from the ISY's device "type"

Now we automatically know which sensors are moisture, motion, and openings! (We also reverse the moisture sensor state, because Insteon reports ON for dry on the primary node.)

* Code review tweaks

The one material change here is that the event subscribers were moved to the `async_added_to_hass` method, as the handlers depend on things that only exist after the entity has been added.

* Handle cases where a sensor's state is unknown

When the ISY first boots up, if a battery-powered sensor has not reported in yet (due to heartbeat or a change in state), the state is unknown until it does.

* Clean up from code review

Fix coroutine await, remove unnecessary exception check, and return None when state is unknown

* Unknown value from PyISY is now -inf rather than -1

* Move heartbeat handling to a separate sensor

Now all heartbeat-compatible sensors will have a separate `binary_sensor` device that represents the battery state (on = dead)

* Add support for Unknown state, which is being added in next PyISY 

PyISY will report unknown states as the number "-inf". This is implemented in the base ISY994 component, but subcomponents that override the `state` method needed some extra logic to handle it as well.

* Change a couple try blocks to explicit None checks

* Bump PyISY to 1.1.0, now that it has been published!

* Remove -inf checking from base component

The implementation of the -inf checking was improved in another branch which has been merged in to this branch already.

* Restrict negative-node and heartbeat support to known compatible types

Not all Insteon sensors use the same subnode IDs for the same things, so we need to use different logic depending on device type. Negative node and heartbeat support is now only used for leak sensors and open/close sensors.

* Use new style string formatting

* Add binary sensor detection for pre-5.x firmware

Meant to do this originally; writing documentation revealed that this requirement was missed!

balloob committed:
Disable html5 notify dependency (#11135)

danielperna84 committed:
Upgrade Homematic (#11149)

* Update pyhomematic

* Update pyhomematic

pvizeli committed:
Homematic next (#11156)

* Cleanup logic & New gen of HomeMatic

* fix lint

* cleanup

* fix coverage

* cleanup

* name consistenc

* fix lint

* Rename ip

* cleanup wrong property

* fix bug

* handle callback better

* fix lint

* Running now

pvizeli committed:
Resolve hostnames (#11160)

balloob committed:
Update frontend

emlove committed:
Perform logbook filtering on the worker thread (#11161)

balloob committed:
Don't connect to cloud if subscription expired (#11163)

* Final touch for cloud component

* Fix test

pvizeli committed:
Add install mode to homematic (#11164)

cmsimike committed:
Remove logging (#11173)

An error was being log that seems more like debug info

mtreinish committed:
Fix X10 commands for mochad light turn on (#11146)

* Fix X10 commands for mochad light turn on

This commit attempts to address issues that a lot of people are having
with the x10 light component. Originally this was written to use the
xdim (extended dim) X10 command. However, not every X10 dimmer device
supports the xdim command. Additionally, it turns out the number of
dim/brighness levels the X10 device supports is device specific and
there is no way to detect this (given the mostly 1 way nature of X10)

To address these issues, this commit removes the usage of xdim and
instead relies on using the 'on' command and the 'dim' command. This
should work on all x10 light devices. In an attempt to address the
different dim/brightness levels supported by different devices this
commit also adds a new optional config value, 'brightness_levels', to
specify if it's either 32, 64, or 256. By default 32 levels are used
as this is the normal case and what is documented by mochad.

Fixes #8943

* make code more readable

* fix style

* fix lint

* fix tests

PhracturedBlue committed:
convert alarmdecoder interface from async to sync (#11168)

* convert alarmdecoder interface from async to sync

* Convert he rest of alarmdecoder rom async to sync

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

* Update alarmdecoder.py

aosadchyy committed:
Rebase to latest. Merged with pull request #11029

pvizeli commented:
This branch is defect. Please make a new PR with clean changes
